File: frontend/src/types/weather.ts
export interface Location {
    id: number;
    name: string;
    country: string;
    latitude: number;
    longitude: number;
    display_name: string;
    is_favorite: boolean;
    created_at: string;
    updated_at: string;
}

export interface LocationSearchResult {
    name: string;
    country: string;
    state?: string;
    latitude: number;
    longitude: number;
    display_name: string;
}

export interface WeatherSnapshot {
    temperature: number;
    feels_like: number;
    temp_min: number;
    temp_max: number;
    pressure: number;
    humidity: number;
    weather_main: string;
    weather_description: string;
    weather_icon: string;
    wind_speed: number;
    wind_deg: number;
    clouds: number;
    visibility: number;
    timestamp: string;
    api_timestamp: number;
}

export interface ForecastItem {
    forecast_timestamp: number;
    temperature: number;
    feels_like: number;
    temp_min: number;
    temp_max: number;
    pressure: number;
    humidity: number;
    weather_main: string;
    weather_description: string;
    weather_icon: string;
    wind_speed: number;
    wind_deg: number;
    clouds: number;
    pop: number;
}

export interface ForecastConfidence {
    label: string;
    score: number;
    reason: string;
}

export interface DailyImpactScores {
    commute: number;
    outdoor: number;
    laundry: number;
    running: number;
}

export interface InsightTimelineEvent {
    timestamp: number;
    title: string;
    severity: string;
    detail: string;
}

export interface ForecastChangeSummary {
    headline: string;
    temperature_delta: number;
    rain_delta: number;
    wind_delta: number;
}

export interface WeatherInsights {
    briefing: string;
    confidence: ForecastConfidence[];
    impact_scores: DailyImpactScores;
    timeline: InsightTimelineEvent[];
    alerts: string[];
    change_summary: ForecastChangeSummary | null;
}

export interface WeatherData {
    location: Location;
    current: WeatherSnapshot | null;
    forecast: ForecastItem[] | null;
    last_synced: string | null;
    insights: WeatherInsights | null;
    sync_note?: string | null;
}

export interface LocationWeatherOverview {
    location: Location;
    current: WeatherSnapshot | null;
    last_synced: string | null;
}

export interface SystemStatus {
    total_locations: number;
    synced_locations: number;
    failed_sync_last_24h: number;
    last_success_sync: string | null;
    sync_interval_seconds: number;
    api_configured: boolean;
}

File: src/schemas/weather.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime

class LocationBase(BaseModel):
    name: str
    country: str
    latitude: float
    longitude: float

class LocationCreate(BaseModel):
    name: str
    country: Optional[str] = None

class LocationUpdate(BaseModel):
    display_name: Optional[str] = None
    is_favorite: Optional[bool] = None

class LocationSearchResult(BaseModel):
    name: str
    country: str
    state: Optional[str] = None
    latitude: float
    longitude: float
    display_name: str

class Location(LocationBase):
    id: int
    display_name: Optional[str]
    is_favorite: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class WeatherSnapshot(BaseModel):
    temperature: float
    feels_like: float
    temp_min: float
    temp_max: float
    pressure: int
    humidity: int
    weather_main: str
    weather_description: str
    weather_icon: str
    wind_speed: float
    wind_deg: Optional[int] = None
    clouds: int
    visibility: Optional[int] = None
    api_timestamp: int
    timestamp: Optional[datetime] = None

class ForecastItem(BaseModel):
    forecast_timestamp: int
    temperature: float
    feels_like: float
    temp_min: float
    temp_max: float
    pressure: int
    humidity: int
    weather_main: str
    weather_description: str
    weather_icon: str
    wind_speed: float
    wind_deg: Optional[int] = None
    clouds: int
    pop: float

class ForecastConfidence(BaseModel):
    label: str
    score: int
    reason: str

class DailyImpactScores(BaseModel):
    commute: int
    outdoor: int
    laundry: int
    running: int

class InsightTimelineEvent(BaseModel):
    timestamp: int
    title: str
    severity: str
    detail: str

class ForecastChangeSummary(BaseModel):
    headline: str
    temperature_delta: float
    rain_delta: float
    wind_delta: float

class WeatherInsights(BaseModel):
    briefing: str
    confidence: List[ForecastConfidence]
    impact_scores: DailyImpactScores
    timeline: List[InsightTimelineEvent]
    alerts: List[str]
    change_summary: Optional[ForecastChangeSummary] = None

class WeatherData(BaseModel):
    location: Location
    current: Optional[WeatherSnapshot] = None
    forecast: Optional[List[ForecastItem]] = None
    last_synced: Optional[datetime] = None
    insights: Optional[WeatherInsights] = None
    sync_note: Optional[str] = None

class LocationWeatherOverview(BaseModel):
    location: Location
    current: Optional[WeatherSnapshot] = None
    last_synced: Optional[datetime] = None

class SystemStatus(BaseModel):
    total_locations: int
    synced_locations: int
    failed_sync_last_24h: int
    last_success_sync: Optional[datetime] = None
    sync_interval_seconds: int
    api_configured: bool

class Preference(BaseModel):
    key: str
    value: str

class PreferenceUpdate(BaseModel):
    value: str
